kiwi/repository/dnf.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) 2015 SUSE Linux GmbH.  All rights reserved.
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import os
+from six.moves.configparser import ConfigParser
+from tempfile import NamedTemporaryFile
+
+# project
+from ..logger import log
+from .base import RepositoryBase
+from ..path import Path
+
+
+class RepositoryDnf(RepositoryBase):
+    """
+    Implements repository handling for dnf package manager
+    """
+    def post_init(self, custom_args=None):
+        """
+        Post initialization method
+
+        Store custom dnf arguments and create runtime configuration
+        and environment
+
+        Attributes
+
+        * :attr:`shared_dnf_dir`
+            shared directory between image root and build system root
+
+        * :attr:`runtime_dnf_config_file`
+            dnf runtime config file name
+
+        * :attr:`command_env`
+            customized os.environ for dnf
+
+        * :attr:`runtime_dnf_config`
+            Instance of ConfigParser
+
+        :param list custom_args: dnf arguments
+        """
+        self.custom_args = custom_args
+        if not custom_args:
+            self.custom_args = []
+
+        # extract custom arguments not used in dnf call
+        if 'exclude_docs' in self.custom_args:
+            self.custom_args.remove('exclude_docs')
+            log.warning('rpm-excludedocs not supported for dnf: ignoring')
```

Comment:
You can do this by doing `--setopt='tsflags=nodocs'`. There is a [pending PR for adding `--nodocs`](https://github.com/rpm-software-management/dnf/pull/748).